Description: Change where the build looks for shared jar files
 Since we strip all of the embedded jar files from the modified upstream source
 we need to change classpaths to point at the installed debianized versions in
 /usr/share/java.
Author: Mat Scales <mat@wibbly.org.uk>
Index: lucene-solr/lucene/contrib/benchmark/build.xml
===================================================================
--- lucene-solr.orig/lucene/contrib/benchmark/build.xml	2011-11-29 09:21:26.029599661 +0000
+++ lucene-solr/lucene/contrib/benchmark/build.xml	2011-11-29 09:21:37.293631182 +0000
@@ -132,8 +132,13 @@
       <pathelement path="${analyzers-common.jar}"/>
       <pathelement path="${facet.jar}"/>
       <path refid="base.classpath"/>
-    	<fileset dir="lib">
-    		<include name="**/*.jar"/>
+    	<fileset dir="/usr/share/java">
+    		<include name="commons-beanutils.jar"/>
+    		<include name="commons-collections3.jar"/>
+    		<include name="commons-compress.jar"/>
+    		<include name="xercesImpl.jar"/>
+    		<include name="commons-digester.jar"/>
+    		<include name="commons-logging.jar"/>
     	</fileset>
     </path>
     <path id="run.classpath">
@@ -174,7 +179,7 @@
 	<path id="collation.runtime.classpath">
 	  <path refid="run.classpath"/>
     <pathelement path="${icu.jar}"/>
-    <fileset dir="${common.dir}/contrib/icu/lib" includes="icu4j*.jar"/>
+    <fileset dir="/usr/share/java" includes="com.ibm.icu-4.4.jar"/>
 	</path>
 	
 	<target name="collation" depends="compile,jar-icu,top-100k-wiki-word-files">
Index: lucene-solr/lucene/contrib/icu/build.xml
===================================================================
--- lucene-solr.orig/lucene/contrib/icu/build.xml	2011-11-29 09:21:26.069599773 +0000
+++ lucene-solr/lucene/contrib/icu/build.xml	2011-11-29 09:21:37.293631182 +0000
@@ -26,7 +26,7 @@
 
 
   <path id="additional.dependencies">
-    <fileset dir="lib" includes="icu4j-*.jar"/>
+    <fileset dir="/usr/share/java" includes="com.ibm.icu-4.4.jar"/>
   </path>
 
   <pathconvert property="project.classpath"
Index: lucene-solr/lucene/contrib/queries/build.xml
===================================================================
--- lucene-solr.orig/lucene/contrib/queries/build.xml	2011-11-29 09:21:26.049599717 +0000
+++ lucene-solr/lucene/contrib/queries/build.xml	2011-11-29 09:21:37.509631787 +0000
@@ -24,7 +24,7 @@
   </description>
 
   <path id="additional.dependencies">
-    <fileset dir="lib" includes="*-oro-*.jar,*-regexp-*.jar"/>
+    <fileset dir="/usr/share/java" includes="regexp.jar"/>
   </path>
 
   <pathconvert property="project.classpath"
Index: lucene-solr/lucene/build.xml
===================================================================
--- lucene-solr.orig/lucene/build.xml	2011-11-29 09:21:25.993599560 +0000
+++ lucene-solr/lucene/build.xml	2011-11-29 09:21:37.509631787 +0000
@@ -227,9 +227,15 @@
       <path id="javadoc.classpath">
         <path refid="classpath"/>
         <pathelement location="${ant.home}/lib/ant.jar"/>
-        <fileset dir=".">
-          <exclude name="build/**/*.jar"/>
-          <include name="**/lib/*.jar"/>
+        <fileset dir="/usr/share/java">
+          <include name="commons-beanutils.jar"/>
+          <include name="commons-collections3.jar"/>
+          <include name="commons-compress.jar"/>
+          <include name="xercesImpl.jar"/>
+          <include name="commons-digester.jar"/>
+          <include name="commons-logging.jar"/>
+          <include name="com.ibm.icu-4.4.jar"/>
+          <include name="regexp.jar"/>
         </fileset>
       </path>
 
Index: lucene-solr/lucene/common-build.xml
===================================================================
--- lucene-solr.orig/lucene/common-build.xml	2011-11-29 09:21:26.009599605 +0000
+++ lucene-solr/lucene/common-build.xml	2011-11-29 09:21:37.509631787 +0000
@@ -138,7 +138,7 @@
   <property name="m2.repository.url" value="file://${maven.dist.dir}"/>
   <property name="m2.repository.private.key" value="${user.home}/.ssh/id_dsa"/>
 
-  <property name="javacc.home" location="${common.dir}"/>
+  <property name="javacc.home" location="/usr/share/java">
   <property name="jflex.home" location="${common.dir}"/>
 
   <path id="jflex.classpath">
@@ -151,8 +151,8 @@
   </path>
 
   <path id="javacc.classpath">
-    <fileset dir="${javacc.home}/">
-      <include name="bin/lib/*.jar"/>
+    <fileset dir="${javacc.home}">
+      <include name="javacc.jar"/>
     </fileset>
   </path>
 
